### ISSUE TITLE
Auto cleanup (LRU)

### ISSUE THREAD

**Issue by ronnyroeller** [PolymerElements/app-storage#46]
We cache the user data in local storage for offline usage. We can't use pure Web Worker because we need to support browsers that don't support it and because the cached data can be changed during offline usage.
Since the space for local storage is limited (depending on the browser), it will sooner or later be filled up. At this point, local storage won't accept new entries.

What's the adviced way to regularly (automatically) clean up local storage (eg LRU) using the app-storage elements?

**Comment by cdata** [comment]
LocalStorage doesn't keep any built-in indication of create / update time. It's possible to keep track of this yourself by recording such values with your data as you change it, but it can be costly to constantly deserialize and re-examine values that are kept in LocalStorage.

`app-localstorage-document` does nothing to improve this condition, nor does it make anything harder than it would be when using the basic Storage API.

`app-indexeddb-mirror` uses IndexedDB as its backing storage layer. This presents an opportunity to index and query by age of record in order to purge old entries. Currently, `app-indexeddb-mirror` does not do this. However, if someone could describe the user story in an issue, we may be able to design a feature of `app-indexeddb-mirror` that supports automatic purging of old data.

So, bottom line: app-storage doesn't do anything to help or hurt this specific use case today. In some cases, it might be able to help in the future. For LocalStorage: I would advise against trying to keep lots of structured data there.

**Comment by ronnyroeller** [comment]
In case somebody else faces the same issue, we implemented and open sourced an element that allows pruning old entries in local storage:
https://github.com/Collaborne/iron-localstorage-behavior

Here the description: https://medium.com/collaborne-engineering/how-to-avoid-local-storage-from-overrunning-4c9702681290#.lwjkb9s9p

**Comment by cdata** [comment]
Thanks @ronnyroeller , looks cool!
